designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,98 @@
+# Search CLI Tool for NuGet.exe and Dotnet.exe
+
+* Status: **Implementing**
+* Author(s): [Advay Tandon](https://github.com/advay26)
+* Issue: [9704](https://github.com/NuGet/Home/issues/9704) - Search CLI Tool for NuGet.exe and Dotnet.exe
+
+## Problem Background
+
+Search functionality from the command line for NuGet.exe and the .NET Core CLI has been a feature request from the NuGet open source community for some time. Additionally, the NuGet API is already equipped to handle search requests, so an extension of this service to command line users would be both practical to implement and also highly valuable.
+
+## Who are the customers
+
+All NuGet.exe and .NET CLI users.
+
+## Goals
+
+* Providing search functionality for NuGet packages through the commandline
+* Displaying the search results to users in a way that maximizes the visibility of the most relevant results.
+
+## Solution
+
+Th search feature leverages the search service provided by the NuGet API. Search targets multiple sources and displays all of their results one after the other. The command line interface is simple: __./NuGet.exe search \<query\>__.
+```
+PS C:\> ./NuGet.exe search logging
```

Comment:
How is the experience different for dotnet users? The way parameters are passed in to both CLIs is different, for example.

Should we, though, do dotnet support as "future work"?

Reply:
In the latest commit, I have put down creating a prototype for dotnet.exe as a goal, and integrating Search into dotnet.exe as something for 'Future Work'